### ISSUE TITLE
No Metadata Available

### ISSUE THREAD

**Issue by dmj222** [tlarsendataguy/graphyx#20]
![image](https://user-images.githubusercontent.com/132037090/235054196-6781c7bb-8f38-4109-83a8-4566518b1f0e.png)

We are just starting to use the tools.  Using the input tool we get the following on pressing "verify":

"The query was successful but no records were returned.  No metadata is available to generate output fields."

Assume this means we cannot use the "add field" facility, and that it cannot read the meta data from Neo4j (v4.4.8 enterprise)  - but I'm not sure.
Its just a test query: MATCH (t:Team) RETURN t.name and works ok in Neo4j browser.  We are using a specific database, not a default.
I'm not sure what could be wrong.

**Comment by dmj222** [comment]
I can get the tool to work if remove the database name (there is some data in our default database)

**Comment by tlarsendataguy** [comment]
Can you confirm the version of Graphyx that is installed?

The message suggests there are no Team nodes in your nba database. How many records does the query return in browser?

Once I know the version of Graphyx you have, I'll spin up a test instance and do some testing on my side to try and recreate the issue.

**Comment by dmj222** [comment]
Yes, running Graphyx v1.0.6 (downloaded last Friday).  There were a few records returned.  See below for a new simple query.  When I tried using the default database (database left blank in graphyx tool) I could get a (different, but similar) query to work. But still can't get the specific database we are using to work when validate button is pressed.  See following screen shots of browser results and graphyx settings.
![image](https://user-images.githubusercontent.com/132037090/235383786-2517d780-932f-4a65-bf08-66bb48fb5597.png)

![image](https://user-images.githubusercontent.com/132037090/235383836-084c8d38-e3a6-4bc3-9d59-28d03d0fc188.png)

**Comment by tlarsendataguy** [comment]
I'm still looking into this, but so far I have been unable to replicate.

Are you familiar with Alteryx's HTML Developer Tools?

**Comment by dmj222** [comment]
No, I'm not familiar with those.  Did you try with a non default DB?  I tried some different cypher queries for the input tool but all fail in the same way.  I also tried same connection, but for the default DB (i.e. blank in tool db field) and they all work, when cypher returns a non-null value.  I as mentioned, I can use the export tool to load data into neo4j in that same "cedb" database.  Note, the database name is all lowercase chars in neo4j.

**Comment by tlarsendataguy** [comment]
Yes, I frequently use the input tool with non-default DBs. I remember a version of the tools had the 'neo4j' database name embedded in the HTTP request, so the custom DBs weren't working. But that was from last year and should not be present in 1.0.6.

I will be updating the tools to 1.0.7 today or tomorrow. If ok with you, let's wait until it's available and see if it fixes your issue.

**Comment by dmj222** [comment]
That's fine thanks

**Comment by tlarsendataguy** [comment]
Apologies for the delay on this - the day job caught fire and required my attention.

I just released version 2.0.0 of the connectors. Try removing version 1 and installing version 2.

Note that version 2 is a breaking change. The URL you need to specify is no longer the HTTP endpoint, but rather the bolt endpoint. For you, that might be something like `bolt://dai.mip.com.au:7687`.

Let me know if you continue receiving the error and we can troubleshoot.
